challenge-10/challenge-10.js

```diff
@@ -1,113 +1,207 @@
+    */
+    function showOperationMessage( operator, number1, number2 ){
+        return ('A operação ' + number1 + ' ' + operator + ' ' + number2 + ' =');
+    }
 
-/*
-Crie uma função chamada "showOperationMessage" que recebe três parâmetros:
-- o operador, o primeiro número e o segundo número. O retorno da função
-deve ser a frase:
-'A operação [NUMBER1] [OPERATOR] [NUMBER2] =';
-Essa função mostrará a mensagem da operação que criaremos mais abaixo.
-*/
-// ?
+    /*
+    Crie uma função chamada "showErrorMessage" que recebe um parâmetro: o
+    operador da operação cálculo, quando a operação não for válida.
+    Essa função deverá retornar a frase:
+    'Operação "[OPERATOR]" não permitida!'
+    */
+    function showErrorMessage( operator ){
+        return 'Operação ' + operator + ' não é permitida';
+    }
 
-/*
-Crie uma função chamada "showErrorMessage" que recebe um parâmetro: o
-operador da operação cálculo, quando a operação não for válida.
-Essa função deverá retornar a frase:
-'Operação "[OPERATOR]" não permitida!'
-*/
-// ?
+    /*
+    Nossa calculadora está pronta! Agora vamos testá-la:
+    PASSO 1:
+    - Declare 3 variáveis: "number1" e "number2", iniciando com valor zero, e
+    "operationSignal", sem valor por enquanto.
+    */
+    var number1 = 0;
+    var number2 = 0;
+    var operationSignal;
 
-/*
-Nossa calculadora está pronta! Agora vamos testá-la:
-PASSO 1:
-- Declare 3 variáveis: "number1" e "number2", iniciando com valor zero, e
-"operationSignal", sem valor por enquanto.
-*/
-// ?
+    /*
+    PASSO 2:
+    Atribua à variável operationSignal o operador de soma, e declare uma
+    variável chamada "sum", que receba a função "calculator", passando por
+    parâmetro a variável que recebeu o sinal da operação.
+    */
+    operationSignal = '+';
+    var sum = calculator(operationSignal);
 
-/*
-PASSO 2:
-Atribua à variável operationSignal o operador de soma, e declare uma
-variável chamada "sum", que receba a função "calculator", passando por
-parâmetro a variável que recebeu o sinal da operação.
-*/
-// ?
+    /*
+    PASSO 3:
+    "sum" agora é uma função, e, se o sinal correto não foi passado para a
+    função "calculator", "sum" será false. Certifique-se de que "sum" não é
+    "false", e então atribua às variáveis "number1" e "number2", dois números
+    que serão os operandos da operação de soma.
+    Após isso, mostre no console o resultado da operação, passando dois
+    parâmetros para o método "log" de "console":
+    - O primeiro será a mensagem da operação (usando a função criada acima);
+    - O segundo, a função de soma, passando os dois operandos.
+    - Se "sum" for "false", mostrar no console a mensagem de erro.
+    */
+    if(sum){
+        number1 = 4;
+        number2 = 6;
+        console.log( showOperationMessage( operationSignal, number1, number2 ), sum( number1, number2) );
+    }
+    else{
+        console.log( showOperationMessage( operationSignal ));
```

Comment:
-1
Aqui no `else` só vai cair se `sum` for `false`. Nesse caso, você deveria usar a função `showErrorMessage` para mostrar o erro :)